docusafe2-business-impl/src/main/java/de/adorsys/docusafe2/business/impl/dfscredentialservice/DFSCredentialServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package de.adorsys.docusafe2.business.impl.dfscredentialservice;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import de.adorsys.dfs.connection.api.complextypes.BucketPath;
+import de.adorsys.dfs.connection.api.domain.Payload;
+import de.adorsys.dfs.connection.api.service.api.DFSConnection;
+import de.adorsys.dfs.connection.api.service.impl.SimplePayloadImpl;
+import de.adorsys.dfs.connection.impl.factory.DFSConnectionFactory;
+import de.adorsys.docusafe2.business.api.cmsencryption.CMSEncryptionService;
+import de.adorsys.docusafe2.business.api.dfscredentialservice.DFSCredentialService;
+import de.adorsys.docusafe2.business.api.keystore.KeyStoreService;
+import de.adorsys.docusafe2.business.api.keystore.types.*;
+import de.adorsys.docusafe2.business.api.types.DFSCredentials;
+import de.adorsys.docusafe2.business.api.types.DocumentContent;
+import de.adorsys.docusafe2.business.api.types.UserIDAuth;
+import de.adorsys.docusafe2.business.impl.cmsencryption.services.CMSEncryptionServiceImpl;
+import de.adorsys.docusafe2.business.impl.keystore.service.KeyStoreServiceImpl;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.cms.CMSEnvelopedData;
+import org.bouncycastle.cms.CMSException;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.PublicKey;
+
+@Slf4j
+public class DFSCredentialServiceImpl implements DFSCredentialService {
+
+    public static final String SYSTEM_DFS_DIRECTORY_NAME = "system-dfs";
+    private static Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    private CMSEncryptionService cmsEncryptionService = new CMSEncryptionServiceImpl();
+    private KeyStoreService keyStoreService = new KeyStoreServiceImpl();
+    private DFSConnection dfsConnectionService = DFSConnectionFactory.get();
+
+    @Override
+    public DFSCredentials getDFSCredentials(UserIDAuth userIDAuth) {
+        log.trace("get dfs credential for user: " + userIDAuth.getUserID().getValue());
+
+        KeyStoreAccess keyStoreAccess = getKeyStoreAccess(userIDAuth);
+
+        BucketPath bucketPathForNewUserInSystemDfs = new BucketPath(SYSTEM_DFS_DIRECTORY_NAME, userIDAuth.getUserID().getValue());
+        byte[] data = dfsConnectionService.getBlob(bucketPathForNewUserInSystemDfs).getData();
+        DocumentContent decrypt = null;
+        try {
```

Comment:
Do not change an Exception. And do not wrap it unless necessary. So this try is useless.